app/controllers/evidence_controller.rb

```diff
@@ -115,7 +115,13 @@ def destroy
       if @evidence.destroy
         track_destroyed(@evidence)
         format.html {
-          redirect_to [current_project, @node],
+          path =
```

Comment:
Wouldn't a `redirect_back` call work in this context?

https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back

Reply:
This sounds like the way to go, thanks.

I tried it but I found the following problem:
Evidence can be deleted from 3 places:
1.  from the issue evidence tab
2. from the node evidence tab
3. from the evidence show page itself (under node)
When using `redirect_back` in case 3, we find an `evidence not found` error, since the evidence does not exist anymore.

So I reverted the change, using again the `back_to` param in delete links.

@aapomm do you see a more elegant way to deal with this?

Reply:
I'd still use `redirect_back` (check the source) and set `request.headers["Referer"]` to `nil` if they match the `Evidence#show` URL for the recently deleted, that way it'll go to the fallback.

I'd even prefer an:

```
if request.headers["Referer"] == Evidence#show
  redirect_to node
else
  redirect_back
end
```

Than having to add (and remember to add in the future) the extra `:back_to` param